### DISCUSSION TITLE
custom http codes on hooks

### DISCUSSION THREAD

**Discussion by dankeyy** [pocketbase/pocketbase#3237] [Q&A]
so i have a collection, on which i have a OnRecordAfterCreateRequest hook that contains some logic operating on the file that was uploaded in that create request.
What I want is to be able to return some custom error code in cases where the file was empty, or another code incase it was invalid, etc
What is the recommended way to approach this?

**Comment by ganigeorgiev** [comment]
I'm not sure if that is what you are looking for but you can try to return `apis.ApiError`.
For examples you can check https://pocketbase.io/docs/go-routing/#error-response.

**Reply by ganigeorgiev** [reply]
_Note that `OnRecordAfterCreateRequest` executes after a successful record create submission and returning an error in the after hook will not revert the already inserted record entry. If you want to prevent the record insertion, you can use the before hook._
